examples/cognitoidp/userpoolclient.yaml

```diff
@@ -1,28 +1,45 @@
----
 apiVersion: cognitoidp.aws.upbound.io/v1beta1
 kind: UserPool
 metadata:
+  annotations:
+    meta.upbound.io/example-id: cognitoidp/v1beta1/userpoolclient
   labels:
-    testing.upbound.io/example-name: example
-  name: example
+    testing.upbound.io/example-name: userpoolclient
+  name: userpoolclient
 spec:
   forProvider:
-    name: example
+    name: userpoolclient
     region: us-west-1
 
 ---
+apiVersion: cognitoidp.aws.upbound.io/v1beta1
+kind: UserPoolClient
+metadata:
+  annotations:
+    meta.upbound.io/example-id: cognitoidp/v1beta1/userpoolclient
+  labels:
+    testing.upbound.io/example-name: userpoolclient
+  name: userpoolclient
+spec:
+  forProvider:
+    name: userpoolclient
+    region: us-west-1
+    userPoolIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: userpoolclient
 
+---
 apiVersion: cognitoidp.aws.upbound.io/v1beta1
 kind: UserPoolClient
 metadata:
   labels:
-    testing.upbound.io/example-name: example
-  name: example
+    testing.upbound.io/example-name: userpoolclient
+  name: userpoolclient-with-dashes
 spec:
   forProvider:
-    name: example
+    name: name-that-doesnt-match-id-regex
```

Comment:
@turkenf I thought it worthwhile to have examples with names that both passed and failed the id regex, as they each displayed different problematic behavior before my bugfix (orphaned resources vs sync errors). Now that bug is fixed, but it makes sense to me to leave this around as a regression test.

I thought just combining both in the same file would work well, but then I'm not quite sure what to do about the example-id annotation. As I understand it, Uptest looks for a match between that annotation and the gvk of the resource to determine "is this the root resource of the example". Since this example has two resources with the same GVK as the "root", I thought maybe I should leave the example id off the second one? What do you think?

I'm happy to do whatever you prefer here.

Reply:
The purpose of using the example-id annotation was to show resources with their dependencies in the marketplace. But even if this approach is not implemented now, we want our samples to be ready if we continue to show resources with their dependencies on the marketplace in the future.

I think the example containing dash/dashes will be sufficient here as it covers both cases, so we can continue with an example containing dash/dashes in its name, what do you think?

Reply:
> I think the example containing dash/dashes will be sufficient here as it covers both cases

I don't understand what you mean. Before [the bugfix](https://github.com/upbound/provider-aws/pull/1021), security group rules with a `spec.forProvider.name` that matched the id regex created a new user pool client each time the crossplane provider pod restarted, while rules with a name that didn't match the regex refused to create user pool clients at all.

Reply:
> I don't understand what you mean. Before https://github.com/upbound/provider-aws/pull/1021, security group rules with a spec.forProvider.name that

I think you mean `UserPoolClient` above, so let me ask, does the `spec.forProvider.name: userpoolclient` example cover a situation that the `spec.forProvider.name: name-that-doesnt-match-id-regex` example does not? If there is no special error situation covered by the example `spec.forProvider.name: userpoolclient`, the example `spec.forProvider.name: name-that-doesnt-match-id-regex` will be sufficient. If not, let's add the two examples separately as in the previous PR.

Reply:
:facepalm: Yes, by security group rule I mean user pool client. Thanks for knowing what I mean even when it's not what I say :wink:.

Yes, there is a category of bugs related to the external name of the UserPoolClient which will be triggered by `spec.forProvider.name: userpoolclient` and which will not be triggered by `spec.forProvider.name: name-that-doesnt-match-id-regex`. I _think_ that any regression that would cause one to fail would also cause the other to fail (although in a different way), but I'm not certain of this and would rather just leave them both in, especially as this is a quick and cheap resource to create/destroy, unlike say a database, or a kafka cluster, or even an ec2 instance.